### DISCUSSION TITLE
How to collect resulting `ids` from an `Model::insert_many()`

### DISCUSSION THREAD

**Discussion by V-FEXrt** [SeaQL/sea-orm#2191] [Q&A] [answered]
Hi all,

I've been poking the docs and code for most of the day and I think I've finally hit a wall. Any tips on how to get the ids out of a `insert_many()` query?

I am using Postgres so `returning` based features should be enabled.

From the docs and poking I've arrived 

```rust
let res: = Fruit::insert_many([apple, orange])
    // Use returning to get the values back from the db
    .exec_with_returning(db)
    .await?;
``` 
But in that scenario `res` is of type `Model` not the `Vec<Model>` that I'd expect. Any idea what I'm doing wrong here?

**Comment by V-FEXrt** [comment]
https://github.com/SeaQL/sea-orm/blob/master/tests/returning_tests.rs#L145 shows an example for `UpdateMany` but I can't seem to get this to work. 

My understanding is that https://github.com/SeaQL/sea-orm/pull/1021 should be the insert version of the above support but no dice.

https://github.com/SeaQL/sea-orm/blob/master/tests/returning_tests.rs#L52 is another example but that is only for the singular `insert` instead of `insert_many`

**Reply by V-FEXrt** [reply]
https://github.com/SeaQL/sea-orm/issues/183 is another related issue if that's helpful

**Comment by krateng** [comment]
Did you ever resolve this? Currently running into the same issue.

**Reply by V-FEXrt** [reply]
Hi! 

Unfortunately I did not. I used a custom select with raw sql instead!

https://www.sea-ql.org/SeaORM/docs/advanced-query/custom-select/

**Reply by tyt2y3** [reply]
It's added in d3e77869abfaee75a038fc3b77cebb458a756950

**Reply by krateng** [reply]
Amazing :D Thank you!

**Comment by tyt2y3** [comment, marked as answer]
Released! https://github.com/SeaQL/sea-orm/releases/tag/1.1.6

```rust
assert_eq!(
    cakes_bakers::Entity::insert_many([
        cakes_bakers::ActiveModel {
            cake_id: Set(1),
            baker_id: Set(2),
        },
        cakes_bakers::ActiveModel {
            cake_id: Set(2),
            baker_id: Set(1),
        },
    ])
    .exec_with_returning_keys(db)
    .await
    .unwrap(),
    [(1, 2), (2, 1)]
);
```
